core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
         }
       })
+      if (cir.fields.nonEmpty && putFields.isEmpty) {
+        throw PicklingException("No fields are captured. You need a custom pickler to handle this.")
+      }
```

Comment:
This happens because `IRs` picks up all sorts weird things as "fields" as fields, including the parameters to the primary constructor. And to cover it up, `reflectivelyWithoutGetter` does try-catch-NoSuchFieldException-ignore.